BLSQ/iaso
# IA-3401 enketo emoji support

Emoji can used in submissions, calculations in the xlsform, currently it's breaking either transparently removing the emoji from the answer, or visibly enketo integration (invalid char error)

Related JIRA tickets : IA-3401

## Self proofreading checklist


## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes

fixes the way we parse xml and mutate the xml for enketo
to support emoji

## How to test

1. start iaso with enketo, seed

```
docker compose -f docker-compose.yml -f docker/docker-compose-enketo.yml up
docker compose run --rm iaso manage seed_test_data --mode=seed --dhis2version=2.40.4.1
```
2. then create a new form (demo emoji)
3. create a version and upload the [xlsfile](https://github.com/user-attachments/files/16871582/test-emoji.2.xlsx)
4. create a submission
5. edit the submission
6. verify the content


## Print screen / video

![image](https://github.com/user-attachments/assets/37a3ebe1-9f9b-4ba6-9e36-12123a7f80b8)

## Notes

The fix_emoji is just black magic... but works.
Thanks to the one who wrote it.
